snipssonos/services/node/music_playback_service.py

```diff
@@ -18,17 +19,12 @@ def play(self, device, music_item):
             return True
 
     def queue(self, device, music_items):
+        session = FuturesSession()
         self.device = device if device else self.device
 
-        results = list()
-
         for music_item in music_items:
             query_url = self._generate_queue_query(music_item)
-            req = requests.get(query_url)
-            results.append(req.ok)
-
-        return all(results)
-
+            session.get(query_url)
```

Comment:
Here, if we want to retrieve the result of the request, we need to call the `.result()` method.

Reply:
Sure, the question is where and what way so it is clean. The dirty way is to pass Hermes to the service so we publish the end of the session as soon as we request to play the first song and then we can just wait for the other requests to be fulfilled. If we go for this we would not even need async requests.